### DISCUSSION TITLE
可以增量更新索引吗？(translation: Can the index be updated incrementally?)

### DISCUSSION THREAD

**Discussion by bojone** [xhluca/bm25s#25] [Q&A] [answered]
如题。实际情况下有很多场景需要实时增量更新。

如果有可能的话，建议参考whoosh、tantivy等，封装为一个比较完整的全文检索底层库。

**Comment by xhluca** [comment, marked as answer]
Translating your question to english (with Google Translate):

> Can the index be updated incrementally?
> As the title says. In practice, there are many scenarios that require real-time incremental updates.
> 
> If possible, it is recommended to refer to whoosh, tantivy, etc., and encapsulate them into a relatively complete full-text search underlying library.

Please correct if there are mistakes.

To answer, there is already a discussion about incremental updates here: #5 #19 #20 

Feel free to use tantivy if you believe it is a better library for your use case.
